src/main/java/org/opensearch/knn/training/TrainingDataConsumer.java

```diff
@@ -24,23 +28,43 @@
 public class TrainingDataConsumer implements Consumer<List<Float[]>> {
 
     private final NativeMemoryAllocation.TrainingDataAllocation trainingDataAllocation;
+    private final VectorDataType vectorDataType;
 
     /**
      * Constructor
      *
      * @param trainingDataAllocation NativeMemoryAllocation that contains information about native memory allocation.
      */
-    public TrainingDataConsumer(NativeMemoryAllocation.TrainingDataAllocation trainingDataAllocation) {
+    public TrainingDataConsumer(NativeMemoryAllocation.TrainingDataAllocation trainingDataAllocation, VectorDataType vectorDataType) {
         this.trainingDataAllocation = trainingDataAllocation;
+        this.vectorDataType = vectorDataType;
     }
 
     @Override
     public void accept(List<Float[]> floats) {
-        trainingDataAllocation.setMemoryAddress(
-            JNIService.transferVectors(
-                trainingDataAllocation.getMemoryAddress(),
-                floats.stream().map(ArrayUtils::toPrimitive).toArray(float[][]::new)
-            )
-        );
+        long memoryAddress = trainingDataAllocation.getMemoryAddress();
```

Comment:
Please add a unit test to validate the code.

Reply:
ack